core/timeline_api.php

```diff
@@ -101,11 +113,15 @@ function timeline_events( $p_start_time, $p_end_time, $p_max_events, $p_filter =
 						break;
 				}
 				break;
+			case FILE_ADDED:
+				$t_event = new IssueFileAddedTimelineEvent( $t_timestamp, $t_user_id, $t_issue_id );
```

Comment:
One of the advantages of having a single event per file is that we can add the file name here, which would make the event more informative.  Similar to how the tag added event includes the tag name.

Reply:
Added attachment filename to timeline event.